src/dao/user_dao/user_dao.py

```diff
@@ -0,0 +1,38 @@
+import os
+from src.dao.databaseDAO import *
+from src.dto.user.user_dto import User
+class UserDAO(DBDAO):
+
+    table = "user_table"
+
+    def __init__(self) -> None:
+        super().__init__(table_name=self.table)
+        self.session = self.getSession()
+        self.engine = self.getEngine()
+
+    def create_user(self, user):
+        with self.engine.connect() as conn:
+            try:
+                conn.execute(text(f"INSERT INTO  {local_database}.{self.table} (`user_email`, `user_password`, `project_id`) VALUES ('{user.user_email}',  '{user.user_password}',  '{user.project_id}')"))
+                conn.commit()
+                _user = user.to_dict()
+                return _user
+            except Exception as e:
+                return e
+        
+
+    def get_all_user(self):
+        with self.engine.connect() as conn:
+            results = conn.execute(text(f"SELECT * FROM  {local_database}.{self.table}")).all()
+            list = []
+            for it in results:
+                user = User(it._mapping["user_email"], it._mapping["user_password"], it._mapping["project_id"])
+                list.append(user.to_dict())
```

Comment:
Line 29 - My DAO will map to DTO first. 
Line 30 - MY DTO object will transform to dictionary object before returning to controller. Hence later my controller will just jsonify(DTO) will do... 
@sweiyang

Reply:
and if it._mapping["user_email] is None?

Reply:
technically it should have value from database because database is source of truth and user_email is unique... 
1. When i create user before saving into DB. Make sure they have values
2. Otherwise i have to do some error handling here.

Reply:
Because all these values are required in my create request, so DB minimally will have empty string. Empty string it can map properly.